.github/workflows/tests.yaml

```diff
@@ -118,6 +118,14 @@ jobs:
           python -V
           conda info
 
+      - name: Code Linting
+        if: matrix.os == 'ubuntu-latest'
+        shell: bash
+        run: |
+          source activate test
+          python -m pip install pre-commit
+          pre-commit run --show-diff-on-failure --all-files
```

Comment:
@snowman2 is there any advantage to making a separate workflow for this?

Reply:
I think that makes sense. I updated the workflow.

Reply:
I am glad you asked. I updated it to match what is done in pyproj for the base linting [ref](https://github.com/pyproj4/pyproj/blob/9283f962e4792da2a7f05ba3735c1ed7f3479502/.github/workflows/tests.yaml#L18-L34). The previous version was specific to pylint and it usually needs all of the dependencies installed to work properly, so it was originally placed within the testing workflow to avoid re-installing everything just to run pylint [pyproj example](https://github.com/pyproj4/pyproj/blob/9283f962e4792da2a7f05ba3735c1ed7f3479502/.github/workflows/tests.yaml#L163-L168). But, since we are using flake8, that location in the workflow no longer needs all of the dependencies and can be moved to the top.

Reply:
Excellent!